### ISSUE TITLE
Second message not showing up?

### ISSUE THREAD

**Issue by berlincount** [stve/em-logger#1]
Hello,

I'm not exactly sure why, but on multiple messages I'm bascially never seeing the second (or following) ones ...

``` ruby
require 'rubygems'
require 'eventmachine'
require 'logger'
require 'em-logger'

log = Logger.new(STDOUT)
logger = EM::Logger.new(log)

EM.run do
  logger.info('ohai')
  logger.info('not showing')

  EM.stop
end
```

akotes@akotes-dev:~$ ruby1.8 loggertest.rb 
I, [2013-06-04T16:05:59.162907 #17722]  INFO -- : ohai
akotes@akotes-dev:~$ ruby1.9.1 loggertest.rb 
I, [2013-06-04T16:07:22.285023 #17904]  INFO -- : ohai

eventmachine (1.0.3)
em-logger (0.0.1)
ruby 1.8.7 (2012-02-08 patchlevel 358) [x86_64-linux]
ruby 1.9.3p194 (2012-04-20 revision 35410) [x86_64-linux]

Any ideas?

**Comment by stve** [comment]
em-logger uses deferables to move logging IO out of the reactor and into a separate thread. If I had to guess at what's happening, the reactor is probably stopping before the deferable in the second log statement can complete. Have you tried putting the stop inside a Timer and delaying the stop for a few seconds?

**Comment by berlincount** [comment]
not really, but will try. this is basically an extension of your example code, though; and there has to be something more reliable than a delay. hmmm.

**Comment by berlincount** [comment]
(whoops, the close was an accident)

**Comment by berlincount** [comment]
``` ruby
require 'rubygems'
require 'eventmachine'
require 'logger'
require 'em-logger'

log = Logger.new(STDOUT)
logger = EM::Logger.new(log)

EM.run do
  logger.info('ohai')
  logger.info('now showing')

  EM.add_shutdown_hook {
    queue_processor = Proc.new do |log_message|
      logger.logger.add(log_message.severity, log_message.message, log_message.progname)
    end
    while not logger.logger_queue.empty? do
      logger_message = logger.logger_queue.pop(&queue_processor)
    end
  }
  EM.stop_event_loop
end
```

works as expected - and probably should be default behaviour, albeit implemented differently?

**Comment by stve** [comment]
em-logger should probably add it's own shutdown hook that drains the queue. More or less that's what you are doing with what you've implemented above.  It's certainly less than ideal to have to add that much code on your side.  What do you think if it were simplified so that you'd be able to do:

``` ruby
EM.add_shutdown_hook { logger.drain }
```

**Comment by berlincount** [comment]
sounds pretty adequate :)

**Comment by stve** [comment]
Took a crack at it and pushed to master, mind giving it a try? 70f673c576941a08f6d5d5b4415b1d279dab27e7

**Comment by berlincount** [comment]
works as expected :) please release!

**Comment by stve** [comment]
Just released v0.1.0, thanks for the suggestion!
